packages/apollo-engine-reporting/src/durationHistogram.ts

```diff
@@ -0,0 +1,57 @@
+export class DurationHistogram {
+  private readonly buckets: number[];
+  public static readonly BUCKET_COUNT = 384;
+  public static readonly EXPONENT_LOG = Math.log(1.1);
+
+  public toArray(): number[] {
+    let bufferedZeroes = 0;
+    const outputArray: number[] = [];
+
+    for (const value of this.buckets) {
+      if (value === 0) {
+        bufferedZeroes++;
+      } else {
+        if (bufferedZeroes === 1) {
+          outputArray.push(0);
+        } else if (bufferedZeroes !== 0) {
+          outputArray.push(-bufferedZeroes);
+        }
+        outputArray.push(value);
+        bufferedZeroes = 0;
+      }
+    }
+    return outputArray;
+  }
+
+  static durationToBucket(durationNs: number): number {
+    const log = Math.log(durationNs / 1000.0);
+    const unboundedBucket = Math.ceil(log / DurationHistogram.EXPONENT_LOG);
+
+    // Comapre <= 0 to catch -0
```

Comment:
Also this catches -Infinity (which is what we get when durationNs = 0). Probably worth noting that. Also typo on "compare"